schema/gks-core/gks-core-source.yaml

```diff
@@ -188,26 +188,40 @@ $defs:
     description: >-
       A structured representation of a code for a defined concept in a terminology or code system.
     properties:
+      code:
+        $ref: '#/$defs/code'
+        description: >-
+          A symbol uniquely identifying the concept, as in a syntax defined by the code system.
+          CURIE format is preferred where possible (e.g. 'SO:0000704' is the CURIE form of the
+          Sequence Ontology code for 'gene'). See https://www.w3.org/TR/curie/.
       label:
         type: string
         description: The human-readable name for the coded concept, as defined by the code system.
       system:
```

Comment:
IMO, `system` is very broad and can lead to ambiguity / inconsistencies since it can be free-text name, IRI, or URL. Have you considered splitting this out into multiple fields (i.e `systemName`, `systemIri`, and `systemUrl`) and making at least of of them required?

Reply:
Oh, just scrolled down

Reply:
Yeah, I kind of like this idea. although we may not need both `systemIri` and `systemUrl` attributes.    

Beyond this, I think we need to have a broader discussion about what it is we want the `Coding` object to do and not to do, and if/how we want to commit to use of CURIEs as values in this object.  And generally how we want to handle enumerations that may be wrapped in `MappableConcepts` and `Codings`.  It still feels a bit complicated and unclear to me.  

The use of CURIEs in the linked data / linkML world provides a way to bypass the need to explicitly list out the system and code separately in the data - and relies on a prefix mapping document that lives outside the data for this.  We are recommending use of CURIEs here in the context of a more FHIR-like `MappableConcept` and `Coding` objects that also capture the system url and code, and now possibly a prefix and prefix expansion.  This all seems duplicative and likely to lead to confusion.  I think we need to tease apart what we want here and implement something clearer to support it.

Reply:
I wonder if there could/should be multiple types of `Coding`? IE for a CURIE vs URL/code.

I appreciate the message compactness that a separate prefix expansion document provides -- it might be good to codify this in some kind of GKS object, though, just so that there's an expected standard for how to access it

Reply:
Yes, this is one option for how to approach the issue, that favors flexibility over interoperability (i.e. enforcing a single way of doing things that would ensure interoperable data) . . . The broader discussion I want to have is what we prioritize in the short and long term to ensure success of the va-spec.  Our initial spec leans more toward flexibility over interoperability - to lower the barrier to early adoption. But in the long run if we true interoperability, we will either need to converge on a single way of doing things, or provide tooling that supports automated transformations between alternate representations that can be used to derive interoperable data from divergent representations. The linkML framework is doing some interesting work in this area that I am keen to explore.